echopype/echodata/combine.py

```diff
@@ -435,14 +446,296 @@ def _check_echodata_channels(
+
+    Returns
+    -------
+    None
+
+    Raises
+    ------
+    ValueError
+        If ``ed_group`` is not ``Vendor_specific``.
+    RuntimeError
+        If non identical filter parameters is found.
+    """
+    if ed_group != "Vendor_specific":
+        raise ValueError("Group must be `Vendor_specific`!")
+
+    if len(ds_append_dims) > 0:
+        # If there's a dataset appending dimension, drop for comparison
+        # of the other values... everything else should be identical
+        ds_list = [ds.drop_dims(ds_append_dims) for ds in ds_list]
+
+    it = iter(ds_list)
+    # Init as identical, must stay True.
+    is_identical = True
+    dataset = next(it)
+    for next_dataset in it:
+        is_identical = dataset.identical(next_dataset)
+        if not is_identical:
+            raise RuntimeError(
+                f"Non identical filter parameters in {ed_group} group. " "Objects cannot be merged!"
+            )
+        dataset = next_dataset
+
+
+def _merge_attributes(attributes: List[Dict[str, str]]) -> Dict[str, str]:
+    """
+    Merges a list of attributes dictionary
+
+    Parameters
+    ----------
+    attributes : list of dict
+        List of attributes dictionary
+        E.g. [{'attr1': 'val1'}, {'attr2': 'val2'}, ...]
+
+    Returns
+    -------
+    dict
+        The merged attribute dictionary
+    """
+    merged_dict = {}
+    for attribute in attributes:
+        for key, value in attribute.items():
+            if value == "" and key not in merged_dict:
+                merged_dict[key] = value
+            elif value != "":
+                merged_dict[key] = value
+    return merged_dict
+
+
+def _capture_prov_attrs(
+    attrs_dict: Dict[str, List[Dict[str, str]]], echodata_filenames: List[str], sonar_model: str
+) -> xr.Dataset:
+    """
+    Capture and create provenance dataset,
+    from the combined attribute values.
+
+    Parameters
+    ----------
+    attrs_dict : dict of list
+        Dictionary of attributes for each of the group.
+        E.g. {'Group': [{'attr1': 'val1'}, {'attr2': 'val2'}, ...]}
+    echodata_filenames : list of str
+        The filenames of the echodata objects
+    sonar_model : str
+        The sonar model
+
+    Returns
+    -------
+    xr.Dataset
+        The provenance dataset for all the attribute values
+
+    """
+    index_keys = [ED_GROUP, ED_FILENAME]
+    df_list = []
+    for group, attributes in attrs_dict.items():
+        df = pd.DataFrame.from_records(attributes)
+        df.loc[:, ED_FILENAME] = echodata_filenames
+        df[ED_GROUP] = group
+        df = df.set_index(index_keys)
+        df_list.append(df)
+
+    df_merged = reduce(
+        lambda left, right: pd.merge(left, right, on=index_keys, how="outer"), df_list
+    )
+    prov_ds = df_merged.to_xarray()
+    # Set these provenance as string
+    if sonar_model.lower() != "ek80":
+        # Skip for EK80
+        # TODO: This is right now problematic with array attribute for filter coeff
```

Comment:
Now that the filter coeffs are data variables in the Vendor-specific group, this would pass?